server/app/mutations/host_nodes/update.rb

```diff
@@ -18,7 +23,63 @@ def execute
 
       notify_grid(self.host_node.grid)
 
-      self.host_node
+      # Update availability if given and trigger appropriate actions
+      if self.availability && self.host_node.availability != self.availability
+
+        self.host_node.set(:availability => self.availability)
+
+        case self.availability
+        when HostNode::Availability::ACTIVE
+          start_stateful_services(self.host_node)
+          # TODO Should this also trigger full re-scheduling or just wait for the next loop to do it?
+        when HostNode::Availability::DRAIN
+          re_deploy_needed_services(self.host_node)
+          stop_stateful_services(self.host_node)
+        end
+      end
+
+      self.host_node.reload
+    end
+
+    # Re-deploy needed services, new deployments will filter out evacuated node
+    def re_deploy_needed_services(host_node)
+      services = find_services(host_node)
+      services.each do |service|
+        GridServiceDeploy.create!(grid_service: service)
```

Comment:
It just makes the drain operation not to start "immediately" as it will happen basically on next scheduling loop.